proxy/proxy_test.go

```diff
@@ -1247,6 +1248,77 @@ func TestFlusherImplementation(t *testing.T) {
 	}
 }
 
+func TestMeasureProxyWatch(t *testing.T) {
+	mo := metrics.Options{
+		Format:                     metrics.PrometheusKind,
+		EnableProxyRequestMetrics:  true,
+		EnableProxyResponseMetrics: true,
+	}
+	m := metrics.NewMetrics(mo)
+	defer m.Close()
+
+	doc := `*  -> latency("10ms") -> backendLatency("20ms") -> status(200) -> <shunt>`
+	tp, err := newTestProxyWithParams(doc, Params{Metrics: m})
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	defer tp.close()
+
+	u, _ := url.ParseRequestURI("https://www.example.org/hello")
+	r := &http.Request{URL: u, Method: "GET"}
+	w := httptest.NewRecorder()
+	tp.proxy.ServeHTTP(w, r)
```

Comment:
While this works, I think we prefer to issue a real http request to the test server via test client.